rules/prefer-module.js

```diff
@@ -198,6 +205,138 @@ const isTopLevelReturnStatement = node => {
 	return true;
 };
 
+const isParentLiteral = node => {
+	if (node?.type !== 'Literal') {
+		return false;
+	}
+
+	return node.value === '.' || node.value === './';
+};
+
+const isImportMeta = node =>
+	node.type === 'MetaProperty'
+	&& node.meta.name === 'import'
+	&& node.property.name === 'meta';
+
+/** @returns {node is import('estree').NewExpression} */
+const isNewURL = node =>
+	node.type === 'NewExpression'
+	&& node.callee.type === 'Identifier'
+	&& node.callee.name === 'URL';
+
+/** @returns {node is import('estree').MemberExpression} */
+const isAccessPathname = node =>
+	node.type === 'MemberExpression'
+	&& getPropertyName(node) === 'pathname';
+
+function isCallNodeBuiltinModule(node, propertyName, nodeModuleNames, sourceCode) {
+	if (node.type !== 'CallExpression') {
+		return false;
+	}
+
+	const visited = new Set();
+
+	return checkExpression(node.callee, 'property');
+
+	/** @param {import('estree').Expression} node */
+	function checkExpression(node, checkKind) {
+		if (node.type === 'MemberExpression') {
+			if (checkKind !== 'property' || getPropertyName(node) !== propertyName) {
+				return false;
+			}
+
+			return checkExpression(node.object, 'module');
+		}
+
+		if (node.type === 'CallExpression') {
+			if (checkKind !== 'module') {
+				return false;
+			}
+
+			// Check process.getBuiltinModule('x')
+			return (
+				isMemberExpression(node.callee, {property: 'getBuiltinModule', object: 'process'})
+				&& isModuleLiteral(node.arguments[0])
+			);
+		}
+
+		if (node.type !== 'Identifier') {
+			return false;
+		}
+
+		if (visited.has(node)) {
+			return false;
+		}
+
+		visited.add(node);
+
+		const variable = findVariable(sourceCode.getScope(node), node);
+		if (!variable || variable.defs.length !== 1) {
+			return;
+		}
+
+		return checkDefinition(variable.defs[0], checkKind);
+	}
+
+	/** @param {import('eslint').Scope.Definition} define */
+	function checkDefinition(define, checkKind) {
+		if (define.type === 'ImportBinding') {
+			if (!isModuleLiteral(define.parent.source)) {
+				return false;
+			}
+
+			const specifier = define.node;
+			return checkKind === 'module'
+				? (specifier?.type === 'ImportDefaultSpecifier' || specifier?.type === 'ImportNamespaceSpecifier')
+				: (specifier?.type === 'ImportSpecifier' && specifier.imported.name === propertyName);
```

Comment:
Seems missing `import * as path ...` ?

Reply:
We already check if it's an `ImportNamespaceSpecifier` when `checkKind === 'module'` 👍

Reply:
Sorry, missed.